### PR TITLE
feat: DecodeJob now returns the error instead of hiding it

### PR DESCRIPTION
**Pull request by josegonzalez**

### CODE REVIEW

File: queue.go [hunk 1]
```diff
@@ -216,8 +216,8 @@ func (q *Queue) getJobInBucketByID(id uint64, bucketName string) (*Job, error) {
 			log.Printf("Unknown key: %d", id)
 			return nil
 		}
-		job = DecodeJob(jobBytes)
-		return nil
+		job, err = DecodeJob(jobBytes)
+		return err
```
Comment (chiefnoah) [inline queue.go:220]: You should be able to just return the result from `DecodeJob` here. At least you need to return both `job` and `err`.

**Comment by chiefnoah** [comment]
My original comment was off, but I tried to merge and it broke so I reverted it. Feel free to fix and re-submit this PR.

**Comment by josegonzalez** [comment]
What exactly broke?

**Comment by chiefnoah** [comment]
https://cloud.drone.io/chiefnoah/goalpost/12/1/2

**Pull request merged**